src/yz_kv.erl

```diff
@@ -141,8 +141,15 @@ get_md_entry(MD, Key) ->
 -spec get_index(bkey()) -> index_name().
 get_index({Bucket, _}) ->
     BProps = riak_core_bucket:get_bucket(Bucket),
+    get_index_from_bucket_props(BProps).
+
+get_index_from_bucket_props(BProps) ->
     proplists:get_value(?YZ_INDEX, BProps, ?YZ_INDEX_TOMBSTONE).
 
+is_search_enabled_for_bucket(BucketProps) ->
+    yz_kv:get_index_from_bucket_props(BucketProps) =/=
```

Comment:
Not super important, but we don't need to qualify this call with the module name - could save a tiny bit of verbosity and remove it if you wanted.